### ISSUE TITLE
Cannot run `] add GNSSReceiver` on a machine which does not support CUDA (i.e. those with AMD GPUs)

### ISSUE THREAD

**Issue by adam-jahraus** [JuliaGNSS/GNSSReceiver.jl#38]
Due to the transient dependency on `CUDA.jl` via `Tracking.jl`, I cannot add `GNSSReceiver.jl` on my machine, as it has an AMD GPU. 

Additionally, `Tracking.jl`'s readme indicates that the GPU implementation is not as performant as the CPU implementation, which to me indicates it's an experimental feature. It would be very helpful if the `CUDA.jl` dependency could be changed to an optional dependency, so that those of us without Nvidia GPUs can still use this package.

**Comment by adam-jahraus** [comment]
This is extra strange, because if I clone `Tracking.jl` locally, I can run the Tests perfectly fine. Further investigation ensuing.

**Comment by adam-jahraus** [comment]
Indeed, when I clone _this_ repo locally, and try to run the tests, I get the error again:

![image](https://github.com/user-attachments/assets/e04143c4-e638-4fbc-836d-169b2998ef0c)

**Comment by adam-jahraus** [comment]
This seems to be some sort of compatibility issue, but it might be due to `GNSSReceiver.jl` not being up to date with the latest `Tracking.jl` (or perhaps `GNSSSignals.jl`)? Or it could just be something in the `Project.toml` that's out of sync, it's not obvious to me.

**Comment by zsoerenm** [comment]
I'm sorry for the bad experience.
Yes, GNSSReceiver isn't yet compatible with the latest changes from Tracking.jl and GNSSSignals.jl
However, it should still be possible if you just add GNSSReceiver. It doesn't matter if your computer has a CUDA graphics card or not. CUDA.jl will recognize that automatically and it will not load.
My guess is that your Julia version is too new and the old CUDA version isn't compatible with that.
So could you try Julia v1.7 or v1.8?

That being sad, I'd like to update GNSSReceiver with latest changes in Tracking.jl soon ;)

**Comment by adam-jahraus** [comment]
Ah good point, I didn't even think to try this with a version of Julia besides 1.11, I just have `juliaup` default to `release`. I'll see if I can narrow down the problem to a particular version. 

If I can figure out a solution, I'll make a PR :)

**Comment by adam-jahraus** [comment]
I can confirm that `GNSSReceiver.jl` does in fact instantiate on Julia v1.9, but not on Julia v1.10 or v1.11.

**Comment by ErikBuer** [comment]
> I'm sorry for the bad experience. Yes, GNSSReceiver isn't yet compatible with the latest changes from Tracking.jl and GNSSSignals.jl However, it should still be possible if you just add GNSSReceiver. It doesn't matter if your computer has a CUDA graphics card or not. CUDA.jl will recognize that automatically and it will not load. My guess is that your Julia version is too new and the old CUDA version isn't compatible with that. So could you try Julia v1.7 or v1.8?
> 
> That being sad, I'd like to update GNSSReceiver with latest changes in Tracking.jl soon ;)

@zsoerenm, could you elaborate on what changes are necessary?
Perhaps open some issues on the necessary changes so others can see the scope and help :)
